core/src/kmx/kmx_plus.cpp

```diff
@@ -0,0 +1,224 @@
+
+#if KMXPLUS_DEBUG
+static void
+dump_section_name(KMX_DWORD ident) {
+  for (int i = 0; i < 4; i++) {
+    putchar((ident & 0xFF000000) >> 24);
+    ident <<= 8;
+  }
+}
+
+static void
+dump_kmxplus_header(const COMP_KMXPLUS_HEADER* hdr) {
+  if (hdr == NULL) {
+    printf("! dump_kmxplus_header: NULL header\n");
+  }
+  dump_section_name(hdr->ident);
+  printf(": (%X) size 0x%X\n", hdr->ident, hdr->size);
+}
+
+static void
+dump_kmxplus_keys(const uint8_t* /*data*/, const COMP_KMXPLUS_KEYS* keys) {
+  if(keys == NULL) {
+    printf("! could not load 'keys' section\n");
+    return;
+  }
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)keys);
+  printf(" count: #0x%X\n", keys->count);
+  for (KMX_DWORD i = 0; i<keys->count; i++) {
+    printf(" #0x%d\n", i);
+    const COMP_KMXPLUS_KEYS_ENTRY& entry = keys->entries[i];
+    printf("  vkey\t0x%X\n", entry.vkey);
+    printf("  mod\t0x%X\n", entry.mod);
+    printf("  flags\t0x%X\n", entry.flags);
+    if (entry.flags & LDML_KEYS_FLAGS_EXTEND) {
+        printf("  \t Extend: String #0x%X\n", entry.to);
+    } else {
+        printf("  \t UTF-32:U+%04X\n", entry.to);
+    }
+  }
+}
+
+static void
+dump_kmxplus_loca(const uint8_t* /*data*/, const COMP_KMXPLUS_LOCA* loca) {
+  if(loca == NULL) {
+    printf("! could not load 'loca' section\n");
+    return;
+  }
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)loca);
+  // TODO-LDML
+  for(KMX_DWORD i=0; i<loca->count; i++) {
+    printf(" Locale #%d: #0x%X\n", i, loca->entries[i].locale);
+  }
+}
+
+static void
+dump_kmxplus_meta(const uint8_t* /*data*/, const COMP_KMXPLUS_META* meta) {
+  if(meta == NULL) {
+    printf("! could not load 'meta' section\n");
+    return;
+  }
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)meta);
+  printf(" name:\t#0x%X\n", meta->name);
+  printf(" author:\t#0x%X\n", meta->author);
+  printf(" conform:\t#0x%X\n", meta->conform);
+  printf(" layout:\t#0x%X\n", meta->layout);
+  printf(" normalization:\t#0x%X\n", meta->normalization);
+  printf(" indicator:\t#0x%X\n", meta->indicator);
+  printf(" settings:\t0x%X\n", meta->settings);
+}
+
+static void
+dump_kmxplus_vkey(const uint8_t* /*data*/, const COMP_KMXPLUS_VKEY* vkey) {
+  if(vkey == NULL) {
+    printf("! could not load 'vkey' section\n");
+    return;
+  }
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)vkey);
+  // TODO-LDML
+}
+
+static void
+dump_kmxplus_strs(const uint8_t* /*data*/, const COMP_KMXPLUS_STRS* strs) {
+  dump_kmxplus_header((const COMP_KMXPLUS_HEADER*)strs);
+  printf("strs: count 0x%X\n", strs->count);
+  for (KMX_DWORD i=0; i<strs->count; i++) {
+    KMX_WCHAR buf[BUFSIZ];
+    printf("#0x%X: ", i);
+    PKMX_WCHAR str = strs->get(i, buf, BUFSIZ);
+    if (!str) {
+        printf("NULL/ERR\n");
+        continue;
+    }
+    for(int j=0; str[j] && j<0x30; j++) {
+        if (str[j] < 0x7F && str[j] != 0x0020 && str[j] > 0x20) {
+            putchar(str[j]);
+        } else {
+            printf("U+%04X ", str[j]);
```

Comment:
```suggestion
            printf(" U+%04X ", str[j]);
```

We'll end up with space being written as `U+0020`, is that your intention? Either way if we mix ascii with other characters, we will either have chars bumping up against U+xxxx (`abcU+0020 U+00C1 `) or excess spaces (`abc U+0020  U+00C1 `) with my fix ... which do you prefer?